src/main/java/io/stargate/sgv2/jsonapi/service/resolver/CreateVectorIndexCommandResolver.java

```diff
@@ -61,18 +61,12 @@ public Operation resolveTableCommand(
     }
 
     if (definitionOptions != null) {
-      if (similarityFunction != null && sourceModel != null) {
-        throw SchemaException.Code.INVALID_INDEX_DEFINITION.get(
-            Map.of(
-                "reason",
-                "Only one of `metric` or `sourceModel` options should be used for `vector` type column"));
-      }
-      if (sourceModel != null && !VectorConstant.SUPPORTED_SOURCES.contains(sourceModel)) {
+      if (sourceModel != null && VectorConstant.SUPPORTED_SOURCES.get(sourceModel) == null) {
         throw SchemaException.Code.INVALID_INDEX_DEFINITION.get(
             Map.of(
                 "reason",
                 "Invalid `sourceModel`. Supported source models are: "
-                    + VectorConstant.SUPPORTED_SOURCES));
+                    + VectorConstant.SUPPORTED_SOURCES.keySet()));
```

Comment:
Invalid 'sourceModel' do we want to also prompt what is wrong, like the value.
Nit.

Reply:
The error messages has the information, it will return as "Provided index configuration is not valid: Invalid `sourceModel`. Supported source models are: ["ada002", "openai_v3_small", "openai_v3_large", "bert", "gecko", "nv_qa_4", "cohere_v3"]"
Are you looking for anything more?

Reply:
Yeah, I mean the wrong input.